lib/variable/include/scipp/variable/transform.h

```diff
@@ -616,6 +651,17 @@ template <bool dry_run> struct in_place {
                         const Other &...other) {
     using namespace detail;
     (scipp::expect::includes(var.dims(), other.dims()), ...);
+
+    if constexpr (!std::is_base_of_v<
+                      core::transform_flags::force_variance_broadcast_t, Op>) {
+      if (const auto dims = merge(var.dims(), other.dims()...);
+          (bad_variance_broadcast(dims, other) || ...))
+        throw_variances_broadcast_error(var, other...);
+      if (is_bins(var) || (is_bins(other) || ...))
+        if (((other.has_variances() && !is_bins(other)) || ...))
+          throw_variances_broadcast_error(var, other...);
+    }
```

Comment:
Duplicate code? If `is_bins(handles)`  works in line 416, it should be possible to extract this code into a separate function.

Reply:
Not sure it is the same, since we are not checking `var` in the nested `if`?